genesyscloud/resource_genesyscloud_responsemanagement_response_test.go

```diff
@@ -0,0 +1,310 @@
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "texts.1.content", textsContent1),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "texts.1.content_type", textsContentTypes[0]),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "interaction_type", interactionTypes[0]),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "substitutions.0.description", substitutionsDescription),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "substitutions.0.default_value", substitutionsDefaultValue),
+					validateValueInJsonAttr("genesyscloud_responsemanagement_response."+responseResource, "substitutions_schema_id", "type", "object"),
+					validateValueInJsonAttr("genesyscloud_responsemanagement_response."+responseResource, "substitutions_schema_id", "properties."+substitutionsSchema+".type", "string"),
+					validateValueInJsonAttr("genesyscloud_responsemanagement_response."+responseResource, "substitutions_schema_id", "required", substitutionsSchema),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "response_type", responseTypes[0]),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "messaging_template.0.whats_app.0.name", templateName),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "messaging_template.0.whats_app.0.namespace", templateNamespace),
+					resource.TestCheckResourceAttr("genesyscloud_responsemanagement_response."+responseResource, "messaging_template.0.whats_app.0.language", "en_US"),
+					resource.TestCheckResourceAttrPair(
+						"genesyscloud_responsemanagement_response."+responseResource, "asset_ids.0",
+						"genesyscloud_responsemanagement_responseasset."+assetResource, "id"),
+				),
+			},
+			{
+				// Import/Read
+				ResourceName:            "genesyscloud_responsemanagement_response." + responseResource,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"substitutions_schema_id", "messaging_template", "response_type"},
+			},
+		},
+		CheckDestroy: testVerifyResponseManagementResponseDestroyed,
+	})
+}
+
+func generateResponseManagementResponseResource(
+	resourceId string,
+	name string,
+	libraryIds []string,
+	interactionType string,
+	schema string,
+	responseType string,
+	assetIds []string,
+	nestedBlocks ...string,
+) string {
+	if interactionType != "" {
+		interactionType = fmt.Sprintf(`interaction_type = "%s"`, interactionType)
+	}
+	if schema != "" {
+		schema = fmt.Sprintf(`substitutions_schema_id = %s`, schema)
+	}
+	if responseType != "" {
+		responseType = fmt.Sprintf(`response_type = "%s"`, responseType)
+	}
```

Comment:
I found a better approach to this is leaving `response_type = %s` in the return string ,and then either passing `nullValue` or `strconv.Quote("<the value>")` into the function. It keeps things a lot more readable and manageable. (Same for the other fields ofc)

Reply:
You right Charlie, it doesn't look better, thanks